--- facebook.xsd	2008-10-19 18:06:01.000000000 -0700
+++ src/main/resources/schema.xsd	2008-10-26 15:48:04.000000000 -0700
@@ -28,7 +28,7 @@
 </xsd:simpleType>
 
 <xsd:simpleType name="uid">
-  <xsd:restriction base="xsd:int" />
+  <xsd:restriction base="xsd:long" />
 </xsd:simpleType>
 
 <xsd:simpleType name="flid">
@@ -1346,12 +1346,21 @@
 <xsd:element name="data_getObjects_response">
   <xsd:complexType>
     <xsd:sequence minOccurs="0" maxOccurs="unbounded">
-      <xsd:element name="data_getObjects_response_elt" type="container" minOccurs="0" maxOccurs="unbounded" />
+      <xsd:element name="container" type="container" minOccurs="0" maxOccurs="unbounded" />
     </xsd:sequence>
     <xsd:attribute name="list" type="xsd:boolean" />
   </xsd:complexType>
 </xsd:element>
 
+<xsd:complexType name="container">
+  <xsd:sequence>
+    <xsd:element name="name" type="xsd:string" />
+    <xsd:element name="assoc_type" type="xsd:int" />
+    <xsd:element name="assoc_info1" type="assoc_object_type" />
+    <xsd:element name="assoc_info2" type="assoc_object_type" />
+  </xsd:sequence>
+</xsd:complexType>
+
 <xsd:element name="data_getObjectProperty_response" type="xsd:string" />
 
 <xsd:element name="data_setObjectProperty_response" type="xsd:boolean" />
@@ -1454,6 +1463,8 @@
 
 <xsd:element name="feed_publishUserAction_response" type="xsd:boolean" />
 
+<xsd:element name="feed_publishTemplatizedAction_response" type="xsd:boolean" />
+
 <xsd:element name="admin_getMetrics_response">
   <xsd:complexType>
     <xsd:sequence minOccurs="0" maxOccurs="unbounded">
